### PR TITLE
Fix #12737: Space Rings draw the same vehicle 4 times

### LINKED ISSUE #12737: Space Rings draw the same vehicle 4 times
**OS:** Windows 10, Linux (Manjaro)
**Version:** 0.3.0
**Commit/Build:** b51e98215

All 4 vehicles draw the passenger riding in vehicle 0 at the animation frame of vehicle 0

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Build a Space Rings
2. Watch the people ride

**Dump file**

**Screenshots / Video:**

**Save game:**

### PR DESCRIPTION
**Pull request by CorySanin**
paint_space_rings_structure always used vehicle 0 for it's vehicle reference

### CODE REVIEW

**Comment by CorySanin** [comment]
Here ya go!

Peek-2020-08-19-23-17_text

**Comment by duncanspumpkin** [comment]
Probably my mistake in the entity refactor

**Comment by CorySanin** [comment]
Actually, it should probably use `vehicleIndex` instead of `segment`, right? Let me fix that.

**Comment by duncanspumpkin** [comment]
Yes vehicle index should be used. Segment index will get the wrong vehicle depending on rotation.

**Comment by CorySanin** [comment]
Good thing we caught that 😉
Rotating around the ride behaves how I expect now. Looks good to me.

**Comment by duncanspumpkin** [comment]
You'll need to add a changelog entry as this bug will have been in the release

File: distribution/changelog.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - Fix: [#12694] Crash when switching ride types with construction window open.
 - Fix: [#12701] Silent NSIS setup flag /S isn't silent, upgrade pop-up appears anyway.
 - Fix: Incomplete loop collision box allowed overlapping track (original bug).
+- Fix: [#12737] Space Rings draw the same vehicle 4 times.
```
Comment (AaronVanGeffen) [inline distribution/changelog.txt:7]: Sorry to be the bearer of bad news, but we keep the numbered fixes above the unnumbered ones.

Comment (CorySanin) [reply distribution/changelog.txt:6]: Oh, alrighty. Not a problem

Comment (CorySanin) [reply distribution/changelog.txt:6]: Oh shoot, didn't see that you also fixed it 😬

**Pull request merged**